cpp/generate.rb

```diff
@@ -593,28 +645,51 @@ def blockFinderFor( name:"", tag_as:"", start_pattern:nil, needs_semicolon: true
     one_scope_resolution = variable_name_without_bounds.maybe(@spaces).maybe(template_call.without_numbered_capture_groups).then(/::/)
     preceding_scopes = newPattern(
         match: maybe(/::/).zeroOrMoreOf(one_scope_resolution).maybe(@spaces),
-        includes: [ :scope_resolution ]
+        includes: [ :scope_resolution_inner ]
         )
-    cpp_grammar[:scope_resolution] = scope_resolution = newPattern(
-        tag_as: "meta.scope-resolution",
-        match: preceding_scopes.then(
-                match: variable_name_without_bounds,
-                tag_as: "entity.name.type.namespace.scope-resolution"
-            ).maybe(@spaces).maybe(
-                template_call
-            ).then(
-                match: /::/,
-                tag_as: "punctuation.separator.namespace.access"
+    scopeResolutionFinder = ->(tag_extension, grammar_name, hidden_grammar_name) do
```

Comment:
It seems that hidden_grammar_name should just be computed form grammar_name.

Reply:
Okay, I did that at first, it just makes it a little weird when you are trying to search for one of the grammar-repo names because it doesn't actually exist anywhere.

The code I had before just appended `_innner` to the name. If you don't think thats confusing I'll add it back

Reply:
maybe `_inner_generated` to make it a bit more clear that its a synthetic tag name.